.github/workflows/test-sample-apps.yml

```diff
@@ -64,15 +66,27 @@ jobs:
         run: |
           echo "### Testing HTTP2_APP app ###"
           yq e '.applications.[].name' "${GITHUB_WORKSPACE}/http2/apps-manifest.yml" | grep -i 'http2' | while read -r HTTP2_APP; do
-              echo "curl -v --http2-prior-knowledge -H 'Connection: close' 'https://$HTTP2_APP.${{ secrets.CF_DOMAIN }}'"
-              curl -v --http2-prior-knowledge -H "Connection: close" "https://$HTTP2_APP.${{ secrets.CF_DOMAIN }}"
+              ROUTE=${{ github.run_id }}-$HTTP2_APP
+              cf8 map-route $HTTP2_APP ${{ secrets.CF_DOMAIN }} --hostname $ROUTE --app-protocol http2
+              echo "curl -v --http2-prior-knowledge -H 'Connection: close' 'https://$ROUTE.${{ secrets.CF_DOMAIN }}'"
+              curl -v --http2-prior-knowledge -H "Connection: close" "https://$ROUTE.${{ secrets.CF_DOMAIN }}"
           done
           echo "Testing GRPC apps"
-          grpcurl go-grpc-test.${{ secrets.CF_DOMAIN }}:443 example.Example.Run
-          grpcurl -proto http2/java-grpc/app/src/main/proto/example.proto java-grpc-test.${{ secrets.CF_DOMAIN }}:443 example.Example.Run 
-          grpcurl -proto http2/node-grpc/example.proto node-grpc-test.${{ secrets.CF_DOMAIN }}:443 Example.Run 
-          grpcurl -proto http2/python-grpc/example.proto python-grpc-test.${{ secrets.CF_DOMAIN }}:443 Example.Run
-          grpcurl -proto http2/ruby-grpc/example.proto ruby-grpc-test.${{ secrets.CF_DOMAIN }}:443 Example.Run
+          ROUTE=${{ github.run_id }}-go-grpc-test
```

Comment:
I don't think that splitting out `ROUTE` keeps this legible enough.

A bit of a weird idea: you could add a function that prepends, e.g.

```shell
g() {
    echo "${{github.run_id}}-${1:?}"
}
```

to then call it as:

```shell
cf8 map-route go-grpc-test ${{ secrets.CF_DOMAIN }} --hostname $(g go-grpc-test) --app-protocol http2
grpcurl $(g go-grpc-test).${{ secrets.CF_DOMAIN }}:443 example.Example.Run
```

It double evaluation but makes the call / target a bit clearer in my opinion.

Reply:
another thing: would dynamically change the `app-manifest.yml` and the route defined there not also work, similar to how you strip the route now?